packages/story-editor/src/app/story/useStoryReducer/reducers/updateElements.js

```diff
@@ -75,48 +75,53 @@ function updateElements(
     return state;
   }
 
-  const { animationLookup, elements: updatedElements } =
-    oldPage.elements.reduce(
-      ({ animationLookup, elements }, element) => {
-        if (!idsToUpdate.includes(element.id)) {
-          return {
-            animationLookup,
-            elements: [...elements, element],
-          };
-        }
-
-        const { animation, ...elem } = updateElementWithUpdater(
-          element,
-          propertiesOrUpdater
-        );
-
-        const animLookup = animation
-          ? { [animation.id]: { ...animation, targets: [elem.id] } }
-          : {};
-
-        return {
-          animationLookup: {
-            ...animationLookup,
-            ...animLookup,
-          },
-          elements: [...elements, elem],
-        };
-      },
-      {
-        animationLookup: {},
-        elements: [],
+  const animationLookup = oldPage.elements.reduce(
+    (animationLookup, element) => {
+      if (!idsToUpdate.includes(element.id)) {
+        return animationLookup;
       }
+      const { animation, ...elem } = updateElementWithUpdater(
+        element,
+        propertiesOrUpdater
+      );
+      const animLookup = animation
+        ? { [animation.id]: { ...animation, targets: [elem.id] } }
+        : {};
+      return {
+        ...animationLookup,
+        ...animLookup,
+      };
+    },
+    {}
+  );
+
+  const updatedElements = oldPage.elements.reduce((elements, element) => {
```

Comment:
💭  If the reducers from `animationLookup` and `updatedElements` are going to run every time `updateElements.js` is run - does this just add an extra loop over the `oldPage.elements`?

I'm just confused on what is gained from making this split?